crates/gpui/src/style.rs

```diff
@@ -643,28 +648,9 @@ impl Style {
 
         if self.is_border_visible() {
             let border_widths = self.border_widths.to_pixels(rem_size);
-            let max_border_width = border_widths.max();
-            let max_corner_radius = corner_radii.max();
-
-            let top_bounds = Bounds::from_corners(
-                bounds.origin,
-                bounds.top_right() + point(Pixels::ZERO, max_border_width.max(max_corner_radius)),
-            );
-            let bottom_bounds = Bounds::from_corners(
-                bounds.bottom_left() - point(Pixels::ZERO, max_border_width.max(max_corner_radius)),
-                bounds.bottom_right(),
-            );
-            let left_bounds = Bounds::from_corners(
-                top_bounds.bottom_left(),
-                bottom_bounds.origin + point(max_border_width, Pixels::ZERO),
-            );
-            let right_bounds = Bounds::from_corners(
-                top_bounds.bottom_right() - point(max_border_width, Pixels::ZERO),
-                bottom_bounds.top_right(),
-            );
-
```

Comment:
After analyzing the code, I found that this was the cause of the unexpected border clipping.

After removing the content_mask from this code and retaining a single border quad, the rendering finally worked correctly.

<img width="612" height="640" alt="image" src="https://github.com/user-attachments/assets/03cc25fc-354a-4087-b3f3-b6f782ed7798" />

I'm a bit unsure if there are scenarios I've missed. However, judging by the code's performance, all four edges are drawn accurately. I suspect this may be because the previous implementation of border dashing no longer required this behavior.